### ISSUE TITLE
wrong colour when second model plots into empty axes

### ISSUE THREAD

**Issue by lukashergt** [williamjameshandley/anesthetic#19]
When plotting 2 models one of which has more parameters than the other, e.g. LCDM and rLCDM, and if we then plot the model with less parameters first including the unshared parameter, then the second model will have the wrong colour in the axes with the unshared parameter.

The colour should be dependent on the call to `plot_2d` and thus on the model. It should not depend on whether any axis is drawn in for the first or subsequent time.

Example code:
```python
fig, axes= ns_lcdm.plot_2d(['A_s', 'n_s', 'r'], label='LCDM')
ns_rlcdm.plot_2d(axes=axes, diagonal_kwargs={'ls': '--'}, label='rLCDM')
leg = get_legend_proxy(fig)
axes['n_s']['n_s'].legend(leg, ['LCDM', 'rLCDM'], bbox_to_anchor=(1, 1), loc='lower left')
```

![image](https://user-images.githubusercontent.com/18258042/60989967-350fb600-a33f-11e9-88ef-e3f68e31937e.png)

Note, how in the above code the second model gets the additional keyword `ls='--'` which shows that rLCDM is drawn in _orange_ in the upper two axes rows and _blue_ in the bottom row.

For now this behaviour can be circumvented by plotting the model with the most parameters first. This won't help in cases where two models each have an additional parameter not shared with the other, though, say e.g. r+LCDM and alpha+LCDM.

**Comment by williamjameshandley** [comment]
Has this issue been resolved by #15 ?

**Comment by lukashergt** [comment]
fixed in #15

**Comment by lukashergt** [comment]
This causes issues again.

In the image below anything labelled `'mp'` should be blue and anything labelled `'pc'` should be orange. However, this clearly does not work for column and row `'x4'`.

```python
mp = MCMCSamples(root="./tests/example_data/mp")
pc = NestedSamples(root="./tests/example_data/pc")
fig = plt.figure(figsize=(10, 10))
fig, axes = make_2d_axes(['x0', 'x1', 'x2', 'x3', 'x4'], fig=fig);
mp.plot_2d(axes, label="mp");
pc.plot_2d(axes, label="pc");
for y, rows in axes.iterrows():
    for x, ax in rows.iteritems():
        ax.legend()
```
![image](https://user-images.githubusercontent.com/18258042/61948207-134d3a80-af9f-11e9-8f23-fc159ea8d3c3.png)
